oak_runtime/src/lib.rs

```diff
@@ -1373,10 +1406,12 @@ impl Runtime {
 
     /// Create a [`RuntimeProxy`] instance for a new Node, creating the new [`NodeId`]
     /// value along the way.
-    fn proxy_for_new_node(self: Arc<Self>) -> RuntimeProxy {
+    fn proxy_for_new_node(self: Arc<Self>, node_name: &str) -> RuntimeProxy {
+        let node_id = self.new_node_id();
         RuntimeProxy {
-            runtime: self.clone(),
-            node_id: self.new_node_id(),
+            runtime: self,
+            node_id,
+            node_name: format!("{}({})", node_name, node_id.0),
```

Comment:
So this "node name" is the node name concatenated with the node id, and only applies to RuntimeProxy, but Runtime still uses the "naked" node name for logging? BTW I think the terminology is confusing, maybe we should come up with a name (no pun intended) for the combined node name + node id string.

Reply:
Good suggestion to use a different name for clarity. How do you feel about `debug_identifier`? Ideally we would share the terminology across nodes & channels, both of which have this mechanism. 

The runtime does use the combined name for logging as well. It uses the combined name from the proxy when instantiating a node. Agreed though that the code doesn't make that exactly clear.

I will look into refactoring it this for to make it clearer. Would you prefer if I did that in this PR? Originally my was to do a refactor PR after this, which would align the naming across channels & nodes, and switch the mechanism from concatenated the name with the id when setting the name to instead implementing a method that returns this combined identifier from using the (then naked) name and id properties.

Reply:
Yeah a separate PR is a good idea. `debug_identifier` sounds sensible, I don't have a better suggestion :)